packages/nimble-components/src/text-field/styles.ts

```diff
@@ -24,6 +24,7 @@ import { styles as errorStyles } from '../patterns/error/styles';
 import { styles as requiredVisibleStyles } from '../patterns/required-visible/styles';
 import { userSelectNone } from '../utilities/style/user-select';
 
+// prettier-ignore
```

Comment:
Could we leverage CSS layers to avoid the reliance on `:not` selectors?

Like how the button has layers, maybe we can add a layer after `disabled` for the `appearsReadonly` state or something:
https://github.com/ni/nimble/blob/6cfd185413ab6e327b55d296813c88c639d545d8/packages/nimble-components/src/patterns/button/styles.ts#L29

That with the normal thought around don't try and share selectors for multiple states, share custom properties across multiple selectors.
Our [cascade guidance](https://github.com/ni/nimble/blob/6cfd185413ab6e327b55d296813c88c639d545d8/packages/nimble-components/docs/css-guidelines.md#prefer-cascade-for-state-changes) may also need to be updated (after the pattern is settled) to put `[disabled][appearance-readonly]` after `[disabled]`

Reply:
Possible the higher specificity of `[disabled][appearance-readonly]` over `[disabled]` means wee won't even need css layers added. Just might mean introducing more design tokens / private css custom properties for the readonly properties to share across the `[readonly]` and `[disabled][appearance-readonly]` selectors in cascade ordering

Reply:
Sure, I can try out some different CSS approaches. I thought I'd start with `:not` since it was the smallest change and see if we thought it was an acceptable use of `:not` before I made a larger change.

Reply:
I like how it turned out! Thanks for taking a look at it 👍